application/features/authentication.feature

```diff
@@ -0,0 +1,135 @@
+Feature: Authentication
+  In order to identify a specific user
+  As an authorized user
+  I need to be able to authenticate a user based on certain credentials
+
+  Background:
+    Given the requester is authorized
+      And the user store is empty
+      And I provide the following valid data:
+        | property     | value                 |
+        | employee_id  | 123                   |
+        | first_name   | Shep                  |
+        | last_name    | Clark                 |
+        | display_name | Shep Clark            |
+        | username     | shep_clark            |
+        | email        | shep_clark@example.org|
+      And I request "/user" be created
+      And a record exists with an employee_id of "123"
+      And the user has a password of "govols!!!"
+
+
+  Scenario: Authenticate a known user with a matching password
+    Given I provide the following valid data:
+        | property  | value       |
+        | username  | shep_clark  |
+        | password  | govols!!!   |
+    When I request "/authentication" be created
+    Then the response status code should be 200
+      And the following data is returned:
+        | property     | value                 |
+        | employee_id  | 123                   |
+        | first_name   | Shep                  |
+        | last_name    | Clark                 |
+        | display_name | Shep Clark            |
+        | username     | shep_clark            |
+        | email        | shep_clark@example.org|
+        | active       | yes                   |
+        | locked       | no                    |
+
+  Scenario: Attempt to authenticate an unknown user
+    Given I provide the following valid data:
+        | property  | value     |
+        | username  | daddy_o   |
+        | password  | govols!!  |
+    When I request "/authentication" be created
+    Then the response status code should be 404
```

Comment:
According to [our current spec](https://github.com/silinternational/idp-id-broker/blob/develop/api.raml#L207), I think the response for unacceptable credentials would be a `422`. I know some of that needs/needed to be refactored, but it might be worth going ahead and writing the tests to match the current spec. Thoughts?